systemFlake.nix

```diff
@@ -24,18 +25,34 @@
 }@args:
 
 let
+  # clean dirty api input / ensure for a minimum that all expected attrs are present
+  validateHostAttrs = {
+      channelName ? null
+    , system ? null
+    , modules ? []
+    , extraArgs ? {}
+  } @ args: { inherit channelName system modules extraArgs; };
```

Comment:
This accounts for the fact that we cannot do at the top:

```nix
{
  defaultHostAttrs ? { 
      channelName ? null
    , system ? null
    , modules ? []
    , extraArgs ? {}
  }
, ...
} :
```

It also brings each host's `profile` in a consistent shape.